src/stack_mitosis/cli.clj

```diff
@@ -52,26 +54,36 @@
   (let [rds (interpreter/client)
         instances (interpreter/databases rds)]
     (when (interpreter/verify-databases-exist instances [source target])
-      (let [tags (interpreter/list-tags rds instances target)
-            plan (plan/replace-tree instances source target
-                                    :restart restart :tags tags)]
-        (cond (:plan options)
-              (do (println (flight-plan (interpreter/check-plan instances plan)))
-                  true)
-              (:iam-policy options)
-              (do (json/pprint (policy/from-plan instances plan))
-                  true)
-              :else
-              (let [last-action (interpreter/evaluate-plan rds plan)]
-                (not (contains? last-action :ErrorResponse))))))))
+      (let [same-vpc (lookup/same-vpc?
+                      (lookup/by-id instances source)
+                      (lookup/by-id instances target))
+            use-restore-snapshot (or (:restore-snapshot options) (not same-vpc))
+            source-snapshot (if use-restore-snapshot
+                              (interpreter/latest-snapshot rds source)
+                              nil)]
```

Comment:
Style issue, and I'll address when switching to keyword args, but `if` without an else clause defaults to return nil on else.